### PR TITLE
Symantec web security service pack long running

### PR DESCRIPTION
**Pull request by israelpoli**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-5375

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+3. Click **Add instance** to create and configure a new integration instance.
+
+    | **Parameter** | **Required** |
+    | --- | --- |
+    | Server URL | True |
+    | User name | True |
+    | Password | True |
+    | Trust any certificate (not secure) | False |
+    | Use system proxy settings | False |
+    | Fetch interval in seconds, (The minimum threshold is 30 seconds) | True |
```
Comment (ShirleyDenkberg) [inline Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/README.md:16]:
```suggestion
    | Fetch interval in seconds. The minimum threshold is 30 seconds. | True |
```

File: Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+category: Analytics & SIEM
+description: Palo Alto Networks Symantec Cloud Secure Web Gateway Event Collector integration for XSIAM.
```
Comment (ShirleyDenkberg) [inline Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.yml:2]:
```suggestion
description: Palo Alto Networks Symantec Cloud Secure Web Gateway Event Collector integration for Cortex XSIAM.
```

File: Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  required: false
+  type: 8
+  section: Connect
+- defaultvalue: 'true'
+  display: Long Running Instance
+  hidden: true
+  name: longRunning
+  type: 8
+  section: Connect
+- display: Fetch interval in seconds, (The minimum threshold is 30 seconds)
```
Comment (ShirleyDenkberg) [inline Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.yml:36]:
```suggestion
- display: Fetch interval in seconds. The minimum threshold is 30 seconds.
```

File: Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector_description.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+## Symantec Cloud Secure Web Gateway Event Collector
+Use this integration to collect operation logs from Symantec Cloud Secure Web Gateway.
+
+In order to use this integration, you need to enter your Symantec Cloud Secure Web Gateway credentials in the relevant integration instance parameters.
+
+
+### To obtain API Key:
+Create a User API Key that serves as the username and password in the Sync API syntax.
+
+1. In Service Mode, select Account Maintenance > MDM, API Keys.
```
Comment (ShirleyDenkberg) [inline Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector_description.md:10]:
```suggestion
1. In Service Mode, select **Account Maintenance** > **MDM, API Keys**.
```

File: Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector_description.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+2. Create a username/password key.
+3. In the API section (bottom half of the screen), click Add API Key.
```
Comment (ShirleyDenkberg) [inline Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector_description.md:12]:
```suggestion
3. In the API section (bottom half of the screen), click **Add API Key**.
```

File: Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector_description.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+4. Define a Username and Password. The username must be unique as it used by the Web Security Service. If the Web Security Service detects any other API with the same name, the service displays an error message and you must define another name.
+5. Click Add.
```
Comment (ShirleyDenkberg) [inline Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector_description.md:14]:
```suggestion
5. Click **Add**.
```

**Comment by ShirleyDenkberg** [comment]
@yuvalbenshalom Doc review completed.

**Comment by cweltPA** [comment]
@ShirleyDenkberg 
Added another readme file to the main pack level, if you could review it as well : 
https://github.com/demisto/content/blob/8d9147040c0927098e18e3d47fde8d818c7ce51d/Packs/SymantecCloudSecureWebGateway/README.md

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
this is a preliminary review. I did not review yet the zip parts and tests.
but there is already points to start working with.

File: Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  dockerimage: demisto/python3:3.10.13.80593
+  longRunning: true
+  script: ""
+  subtype: python3
+  type: python
+tests:
+- No tests (auto formatted)
+marketplaces:
+- marketplacev2
+fromversion: 6.8.0
```
[the remark this answers is not in the document]
Comment (israelpoli) [reply Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.yml]: This is a required field
so I changed it to version 8.2.0 like the other collectors that merged recently

File: Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.py [hunk 8]
```diff
@@ -0,0 +1,687 @@
...
+                    demisto.debug(f"Crashed on limit of api calls: {e}")
+                    time.sleep(client.fetch_interval)
+                    continue
+                else:
+                    demisto.debug(f"Some ERROR: {e=}")
+                    raise e
+            except Exception as err:
+                demisto.debug(f"Some ERROR: {e=} after the error: {err}")
+                raise e
+        except Exception as err:
```
Comment (yuvalbenshalom) [inline Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.py:566]: probably here you sold remove also the temp file, if it existed.

File: Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.py [hunk 9]
```diff
@@ -0,0 +1,687 @@
...
+        # and continue the loop to fetch with the same parameters.
+        if status == "abort":
+            tmp_file_path.unlink()
+            continue
+
+        (
+            time_of_last_fetched_event,
+            new_events_suspected_duplicates,
+            handling_duplicates,
+        ) = extract_logs_and_push_to_XSIAM(last_run_model, tmp_file_path, token_expired)
```
Comment (yuvalbenshalom) [inline Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.py:580]: you sold remove the themp file if there is an exception here.

File: Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.py [hunk 10]
```diff
@@ -0,0 +1,687 @@
...
+    token = ""
+    if status_match := REGEX_FOR_STATUS.search(str(end_file)):
+        status = status_match.groupdict().get("status", "")
+    if token_match := REGEX_FOR_TOKEN.search(str(end_file)):
+        token = token_match.groupdict().get("token", "")
+
+    return status, token
+
+
+def get_file_size(file_path: Path) -> int:
```
Comment (yuvalbenshalom) [inline Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.py:181]: this is not really nesseary.
better use stat().st_size directly/

also when you write something like this(and you did):
file_size = get_file_size(file_path)
you already know form the variable name that it is file size.
so this is just as readable:
file_size = file_path.stat().st_size.

File: Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.py [hunk 11]
```diff
@@ -0,0 +1,687 @@
...
+        return DEFAULT_FETCH_SLEEP
+    if fetch_sleep < DEFAULT_FETCH_SLEEP:
+        demisto.debug(
+            f"Fetch interval is too low, setting it to minimum of {DEFAULT_FETCH_SLEEP} seconds"
+        )
+        return DEFAULT_FETCH_SLEEP
+    return fetch_sleep
+
+
+def get_events_and_write_to_file_system(
```
Comment (yuvalbenshalom) [inline Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.py:112]: why are you passing all params? lets talk about this.

File: Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.py [hunk 12]
```diff
@@ -0,0 +1,687 @@
...
+        )
+
+    start_date, end_date = get_start_and_end_date(None)
+    params: dict[str, Union[str, int]] = {
+        "startDate": start_date,
+        "endDate": end_date,
+        "token": "none",
+    }
+
+    client.get_logs(params)
```
Comment (yuvalbenshalom) [inline Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.py:614]: dose this run in a resemble time?

File: Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.py [hunk 13]
```diff
@@ -0,0 +1,687 @@
...
+            # If the conversion fails,
+            # defaults to the original start date.
+            demisto.debug("time_of_last_fetched_event is not datetime")
+            start_date_for_next_fetch = start_date
+    else:
+        start_date_for_next_fetch = start_date
+    return start_date_for_next_fetch
+
+
+def management_next_fetch(
```
Comment (yuvalbenshalom) [inline Packs/SymantecCloudSecureWebGateway/Integrations/SymantecCloudSecureWebGatewayEventCollector/SymantecCloudSecureWebGatewayEventCollector.py:385]: maybe calculate next fetch?

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
looks good.
see my comments.

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**